webhook/server/server.go

```diff
@@ -327,14 +327,14 @@ func mutateHandler(w http.ResponseWriter, req *http.Request) {
 		resourceRequests := make(map[string]int64)
 
 		/* unmarshal list of network selection objects */
-		networks, _ := parsePodNetworkSelections(netSelections)
+		networks, _ := parsePodNetworkSelections(netSelections, pod.ObjectMeta.Namespace)
```

Comment:
I think there may be an issue here when pod doesn't have namespace explicitly set in the manifest. If that's the case, .Namespace will be an empty string here, which then results in failure to fetch network attachment.

Here is a snippet from official API description for namespace field:

```
Namespace defines the space within each name must be unique. An empty namespace is equivalent to the "default" namespace, but "default" is the canonical representation. Not all objects are required to be scoped to a namespace - the value of this field for those objects will be empty. Must be a DNS_LABEL. Cannot be updated. More info: http://kubernetes.io/docs/user-guide/namespaces
```

I think we should fall back to "default" if final namespace is still empty string after all substitutes.

Reply:
And of course, this suggests we need a unit test case to cover for an empty namespace in manifest.